federated_aws_cli/cache.py

```diff
@@ -68,6 +65,24 @@ def _safe_write(path):
     f.close()
 
 
+@_requires_safe_cache_dir
+def write_aws_shared_credentials(role_arn, credentials):
+    # Create a sha256 of the role arn, so fix length and remove weird chars
+    path = os.path.join(cache_dir, "aws_shared_creds_" + sha256(role_arn.encode("utf-8")).hexdigest())
```

Comment:
Also, since this file will be used by humans it might make sense to have a human usable name.

Also, I'd imagine we would want a single custom shared credential file into which we can store all the roles that users log into. That way a user could either use their standard non federated roles or use this custom shared credential file and have access to all their federated roles.

This could mean that we pick a good filename and then each role inside the file is stored with a distinct profile name based off the account and IAM role name

Reply:
Oh, so we're not having one file per role? If that's the case, I'll basically need to rewrite all of this. Should it go in the usual ephemeral credentials file? I'd hate to blow up anything they already have.